src/dtx/dtx_common.c

```diff
@@ -172,10 +166,7 @@ dtx_batched_commit(void *arg)
 		dbca = d_list_entry(dmi->dmi_dtx_batched_list.next,
 				    struct dtx_batched_commit_args, dbca_link);
 		cont = dbca->dbca_cont;
-		if (dbca->dbca_deregistering != NULL) {
-			dtx_flush_on_deregister(dmi, dbca);
-			goto check;
-		}
+		ds_cont_child_get(cont);
```

Comment:
Why do we need an extra reference count here? Isn't ds_cont_child being held by 'dbca_cont'?

Reply:
dbca::dbca_cont may be released by dtx_batched_commit_deregister().